src/app/components/dynamic-table/dynamic-table.component.ts

```diff
@@ -46,7 +47,12 @@ export class DynamicTableComponent implements OnInit, OnChanges, AfterContentIni
             // Casual case, displaying data
         } else {
             this.displayedColumns = this.schemaFields.map((f: DataSchemaField) => f.name);
-            this.dataSource.data = this.dataRows;
+            const modifiedRows = this.dataRows.map((x: DataRow) => ({
+                ...x,
+                op: operationColumnMapper(x.op),
+                cssClass: setOperationColumnClass(x.op),
+            }));
+            this.dataSource.data = modifiedRows;
```

Comment:
Should not happen in presentation component. Data should come in a form that will not take dynamic table component to think how to present it. It is not it's function to think what is this data.

Reply:
Fixed